### PR TITLE
[2016.11] Merge forward from 2016.3 to 2016.11

### PR DESCRIPTION
**Pull request by rallytime**
Conflicts:
  - salt/modules/win_system.py
  - tests/unit/modules/win_system_test.py

### CODE REVIEW

**Comment by rallytime** [comment]
@vutny Would you be able to look at the test failure here? Specifically `unit.utils.parsers_test.SaltKeyOptionParserTestCase.test_get_log_level_config`? This is failing on my merge forward from your changes yesterday. I spent some time going through the code and the changes here, but there is a refactor in the parsers file that is confounding the setup mocks you've made in the unit tests. It appears to me that the `log_level: info` bug _is_ fixed and working properly, but I am not sure if there is something else going on with the test set up. Since you've been involved in the parsers rewrite, I'd love some help on this one.

Basically, I had to make a change in your test that is currently failing in 342884018bfd6cc06194aec4a21b2ee0a0d9332e because of some keyerrors that were happening in the test. The key errors started because of the change to the `setup_logfile_logger` function rework in 950cf79cd461a81af43b577b0d23658940ad33d4. Since we're not mocking the call to `setup_logfile_logger` the same way anymore, the `self.log_setup.log_level_logfile` value is `warning` and the test fails.

Obviously, I could add something like this `self.log_setup.log_level_logfile = log_level` after the mock to get the test to "pass", but then that negates the `self.assertEqual(self.log_setup.log_level_logfile, log_level)` check and we won't catch any possible future regressions with this test in the `setup_logfile_logger` like we saw with #38388.

@vutny Let me know if this is something you're able to look into. If not, I can keep looking at it.

**Comment by vutny** [comment]
@rallytime I would really like to offer some help here, but would be really busy till Monday inclusively.
I will try to give you quick suggestions in the in-line comments. I hope that will help to nail the error.

File: salt/utils/parsers.py [hunk 1]
```diff
@@ -654,7 +661,9 @@ def __setup_logfile_logger_config(self, *args):  # pylint: disable=unused-argume
             self.config.pop(self._logfile_loglevel_config_setting_name_)
```
Comment (vutny) [inline salt/utils/parsers.py]: I think the above `if` statement may cause the tests failure. It's really a duplicate of the another `if` from `process_log_level_logfile` function, which already does popping the value form config if unset.

Comment (vutny) [inline salt/utils/parsers.py]: Also, the `cli_log_path` below appears to be completely unused, I remember I have removed it in previous revision.

Right after that there is another popping from config for `log_file` key. It should not affect anything, but that's totally unnecessary.

Everything else looks good.

**Review by vutny — CHANGES_REQUESTED** [review]
@rallytime Try to remove the lines I mentioned.

File: salt/utils/parsers.py (hunk 1 above)
Comment (rallytime) [inline salt/utils/parsers.py]: Oh! I see what you mean. Those were good pointers. I figured out how to get this resolved. Thanks!

**Pull request merged**